devito/symbolics/manipulation.py

```diff
@@ -274,3 +275,13 @@ def indexify(expr):
         except AttributeError:
             pass
     return expr.xreplace(mapper)
+
+
+def collect_temp(expr):
```

Comment:
this is super small and used in only one place, so it's better to squash it inside the main collect routine